### ISSUE TITLE
Singularity "containall" does not create /dev/fd symlink, breaks shell redirection, completion etc.

### ISSUE THREAD

**Issue by olifre** [apptainer/singularity#1419]
## Version of Singularity:

2.4

### Expected behavior

```
$ singularity shell -C ubu
Singularity: Invoking an interactive shell within container...

Welcome inside the Ubuntu 18.04 container.
Singularity ubu:~> exec > >(tee -a foo.log)
Singularity ubu:~> 
```
=> All fine.

### Actual behavior

```
$ singularity shell -C ubu
Singularity: Invoking an interactive shell within container...

Welcome inside the Ubuntu 18.04 container.
Singularity ubu:~> exec > >(tee -a foo.log)
bash: /dev/fd/62: No such file or directory
```

### Steps to reproduce behavior

Just call `singularity shell -C`. 
An easy fix is:
```
ln -s /proc/self/fd /dev/
```
It would be cool if `-C` could take care of creating that symlink, having that also fixes some special cases of shell completion.

**Comment by cclerget** [comment]
Nice catch, thank you @olifre ! I will add /dev/stdin, /dev/stdout, /dev/stderr links too

**Comment by olifre** [comment]
@cclerget Thanks! Yes, that's certainly a good idea! :+1:

**Comment by cclerget** [comment]
Fixed by #1420
